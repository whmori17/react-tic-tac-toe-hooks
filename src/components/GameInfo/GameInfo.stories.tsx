import * as React from 'react';
import { GameInfo, GameInfoProps } from '.';

export default {
  title: 'GameInfo',
};

const gameInfoProps: GameInfoProps = {
  status: 'Go to move #1',
  moves: [<p key={'key'}>fake click</p>],
};

export const gameStartMove = () => <GameInfo {...gameInfoProps} />;
