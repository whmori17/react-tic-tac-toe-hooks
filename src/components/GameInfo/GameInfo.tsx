import * as React from 'react';

export interface GameInfoProps {
  status: string;
}

export const GameInfo: React.FC<GameInfoProps> = ({ status, children }) => {
  return (
    <div className="game-info">
      <div>{status}</div>
      <ol>{children}</ol>
    </div>
  );
};

GameInfo.displayName = 'GameInfo';
