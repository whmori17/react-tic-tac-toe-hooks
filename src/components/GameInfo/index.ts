export { GameInfo, GameInfoProps } from './GameInfo';
