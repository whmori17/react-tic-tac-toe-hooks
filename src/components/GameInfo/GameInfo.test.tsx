import * as React from 'react';
import { shallow } from 'enzyme';
import { GameInfo, GameInfoProps } from './GameInfo';

describe('GameInfo', () => {
  const gameInfoProps: GameInfoProps = {
    status: 'Go to move #1',
    moves: [<p key={'key'}></p>],
  };

  it('renders with correct status', () => {
    const wrapper = shallow(<GameInfo {...gameInfoProps} />);

    expect(wrapper.find('div').contains('Go to move #1')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with correct moves', () => {
    const wrapper = shallow(<GameInfo {...gameInfoProps} />);

    expect(wrapper.find('ol').contains(<p key={'key'}></p>)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
