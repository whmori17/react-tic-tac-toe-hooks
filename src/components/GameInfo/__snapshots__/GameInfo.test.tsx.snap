// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo renders with correct moves 1`] = `
<div
  className="game-info"
>
  <div>
    Go to move #1
  </div>
  <ol>
    <p
      key="key"
    />
  </ol>
</div>
`;

exports[`GameInfo renders with correct status 1`] = `
<div
  className="game-info"
>
  <div>
    Go to move #1
  </div>
  <ol>
    <p
      key="key"
    />
  </ol>
</div>
`;
